src/MCPClient/lib/clientScripts/archivematicaCreateMETSMetadataXML.py

```diff
@@ -0,0 +1,265 @@
+        key = tree.xpath("namespace-uri(.)")
+        checked_keys.append(key)
+    if not key or key not in VALIDATION:
+        key = tree.xpath("local-name(.)")
+        checked_keys.append(key)
+    if not key or key not in VALIDATION:
+        raise ValueError(
+            "XML validation schema not found for keys: {}".format(checked_keys)
+        )
+    return VALIDATION[key]
+
+
+class Resolver(etree.Resolver):
+    def resolve(self, url, id, context):
+        url_scheme = urlparse(url).scheme
+        if url_scheme in ("http", "https"):
+            try:
+                response = requests.get(url)
+            except requests.RequestException:
+                return super().resolve(url, id, context)
+            else:
+                return self.resolve_string(response.text, context)
+        else:
+            return super().resolve(url, id, context)
+
+
+def _validate_xml(tree, schema_uri):
+    schema_type = schema_uri.split(".")[-1]
+    parse_result = urlparse(schema_uri)
+    if not parse_result.scheme and schema_uri == parse_result.path:
+        # URI is a local file
+        try:
+            schema_uri = Path(schema_uri).as_uri()
+        except ValueError:
+            return False, [
+                "XML schema local path {} must be absolute".format(schema_uri)
+            ]
+    try:
+        with urlopen(schema_uri) as f:
+            if schema_type == "dtd":
+                schema = etree.DTD(f)
+            elif schema_type == "xsd":
+                schema_contents = etree.parse(f)
+                try:
+                    schema = etree.XMLSchema(schema_contents)
+                except etree.XMLSchemaParseError:
+                    # Try parsing the schema again with a custom resolver
+                    parser = etree.XMLParser()
+                    resolver = Resolver()
+                    parser.resolvers.add(resolver)
+                    with urlopen(schema_uri) as f2:
+                        schema_contents = etree.parse(f2, parser)
+                    schema = etree.XMLSchema(schema_contents)
+            elif schema_type == "rng":
+                schema_contents = etree.parse(f)
+                schema = etree.RelaxNG(schema_contents)
+            else:
+                return False, [
+                    "Unknown XML validation schema type: {}".format(schema_type)
+                ]
+    except etree.LxmlError as err:
+        return False, ["Could not parse schema file: {}".format(schema_uri), err]
+    if not schema.validate(tree):
+        return False, schema.error_log
+    return True, []
+
+
+def _add_validation_event(mets, file_uuid, schema_uri, valid, errors):
+    event_data = {
+        "eventType": "validation",
+        "eventDetail": 'type="metadata"; validation-source-type="'
+        + schema_uri.split(".")[-1]
+        + '"; validation-source="'
+        + schema_uri
+        + '"; program="lxml"; version="'
+        + etree.__version__
+        + '"',
```

Comment:
This part is a bit hard to read. What do you think of splitting it like this:

```
    event_detail = {
        "type": "metadata",
        "validation-source-type": schema_uri.split(".")[-1],
        "validation-source": schema_uri,
        "program": "lxml",
        "version": etree.__version__,
    }
...
        "eventDetail": "; ".join(
            [f'{k}="{v}"' for k, v in event_detail.items()]
        ),
...
```

Reply:
Nice, thanks!